### ISSUE TITLE
input type="date" fails validation when using locale specific methods from localizations/ 

### ISSUE THREAD

**Issue by tyomitch** [jquery-validation/jquery-validation#2359]
https://jqueryvalidation.org/date-method/ says: _This method is deprecated and will be removed in version `2.0.0`.
Please don't use it, since it relies on the `Date` constructor, which behaves very differently across browsers and locales. Use `dateISO` instead or one of the locale specific methods (in localizations/ and additional-methods.js)._

So we're using `localization/methods_pt.js`, and observing that it causes validation of date inputs to always fail, because they are unconditionally validated using the deprecated method, and not its suggested replacements.

### Environment
* `jquery@3.5.1`
* `jquery-validation@1.19.2`
* `jquery-validation-unobtrusive@3.2.11`
* Chrome 86.0.4240.111

### Steps to reproduce
* https://jsfiddle.net/1ua63jx0/
* focus into and out of the date input: it gets a red border, indicating `class="input-validation-error"`
* if `localization/methods_pt.js` is *not* loaded, then the validation succeeds -- even though our actual date format conforms to `methods_pt`

This is because date input value is always in ISO format, regardless of the display format [(reference)](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date)

### Expected behaviour
* `normalizeAttributeRule` should match `date` input type with `dateISO` rule, instead of `date` rule.

**Comment by PatrickHollweck** [comment]
Yep, We are having the same issue. Given an `<input type="date" />` its `.val()` is always in the format `01-01-1970`.

When we activate the localized methods for Germany the date method is changed to the format `01.01.1970` which is the format that is commonly used in Germany, **but** the format of the input element remains the same which causes the validation to always fail...

**Comment by tyomitch** [comment]
@PatrickHollweck out of curiosity, does my one-line fix at #2360 not help?

**Comment by PatrickHollweck** [comment]
@tyomitch For various reasons, I cannot modify the jquery-validation script since it comes from a module that I cannot modify.
Therefore I needed a solution that works without it... Have not tested your change, but if it works it would be really nice if someone could merge it :)

**Comment by tyomitch** [comment]
I think this is wrong, and the problem still persists.
